### PR TITLE
fix cache config migration fails with float setting on mysql

### LINKED ISSUE #45520: Float value in the TTL will break migration
### Describe the bug

If you set a fractional value in the min TTL or ttl ratio you won't be able to migrate to v50

### To Reproduce

1) spin up v49
2) set min ttl or ttl ratio to 0.5
3) move to v50

### Expected behavior

_No response_

### Logs

```
metabase-mysql1       | 2024-07-13 01:01:07,648 ERROR liquibase.changelog :: ChangeSet migrations/001_update_migrations.yaml::v50.2024-06-12T12:33:09::piranha encountered an exception.
metabase-mysql1       | liquibase.exception.DatabaseException: (conn=8) Truncated incorrect INTEGER value: '0.5' [Failed SQL: (1292) UPDATE cache_config
metabase-mysql1       |    SET config = json_object(
metabase-mysql1       |            'multiplier', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-ttl-ratio'), 10),
metabase-mysql1       |            'min_duration_ms', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-min-ttl'), 60000)
metabase-mysql1       |          )
metabase-mysql1       |  WHERE model = 'root' AND
metabase-mysql1       |        model_id = 0 AND
metabase-mysql1       |        strategy = 'ttl' AND
metabase-mysql1       |        (json_extract(config, '$.multiplier') = 0 OR
metabase-mysql1       |         json_extract(config, '$.min_duration_ms') = 0)]
metabase-mysql1       | 	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:470)
metabase-mysql1       | 	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77)
metabase-mysql1       | 	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:179)
metabase-mysql1       | 	at liquibase.executor.AbstractExecutor.execute(AbstractExecutor.java:141)
metabase-mysql1       | 	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1285)
metabase-mysql1       | 	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:755)
metabase-mysql1       | 	at liquibase.changelog.visitor.UpdateVisitor.executeAcceptedChange(UpdateVisitor.java:119)
metabase-mysql1       | 	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:68)
metabase-mysql1       | 	at liquibase.changelog.ChangeLogIterator$2.lambda$run$0(ChangeLogIterator.java:133)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:122)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:252)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:256)
metabase-mysql1       | 	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:89)
metabase-mysql1       | 	at liquibase.command.core.AbstractUpdateCommandStep.lambda$run$0(AbstractUpdateCommandStep.java:110)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:108)
metabase-mysql1       | 	at liquibase.command.core.UpdateCommandStep.run(UpdateCommandStep.java:105)
metabase-mysql1       | 	at liquibase.command.CommandScope.execute(CommandScope.java:217)
metabase-mysql1       | 	at liquibase.Liquibase.lambda$update$0(Liquibase.java:245)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at liquibase.Liquibase.runInScope(Liquibase.java:1419)
metabase-mysql1       | 	at liquibase.Liquibase.update(Liquibase.java:234)
metabase-mysql1       | 	at liquibase.Liquibase.update(Liquibase.java:212)
metabase-mysql1       | 	at liquibase.Liquibase.update(Liquibase.java:194)
metabase-mysql1       | 	at metabase.db.liquibase$migrate_up_if_needed_BANG_$fn__44522.invoke(liquibase.clj:360)
metabase-mysql1       | 	at metabase.db.liquibase$run_in_scope_locked$reify__44518.run(liquibase.clj:325)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at metabase.db.liquibase$run_in_scope_locked.invokeStatic(liquibase.clj:318)
metabase-mysql1       | 	at metabase.db.liquibase$run_in_scope_locked.invoke(liquibase.clj:301)
metabase-mysql1       | 	at metabase.db.liquibase$migrate_up_if_needed_BANG_.invokeStatic(liquibase.clj:349)
metabase-mysql1       | 	at metabase.db.liquibase$migrate_up_if_needed_BANG_.invoke(liquibase.clj:342)
metabase-mysql1       | 	at metabase.db.setup$migrate_BANG_$fn__53458.invoke(setup.clj:84)
metabase-mysql1       | 	at metabase.db.liquibase$do_with_liquibase$f_STAR___44459.invoke(liquibase.clj:140)
metabase-mysql1       | 	at metabase.db.liquibase$do_with_liquibase.invokeStatic(liquibase.clj:143)
metabase-mysql1       | 	at metabase.db.liquibase$do_with_liquibase.invoke(liquibase.clj:131)
metabase-mysql1       | 	at metabase.db.setup$migrate_BANG_.invokeStatic(setup.clj:73)
metabase-mysql1       | 	at metabase.db.setup$migrate_BANG_.doInvoke(setup.clj:55)
metabase-mysql1       | 	at clojure.lang.RestFn.invoke(RestFn.java:425)
metabase-mysql1       | 	at metabase.db.setup$run_schema_migrations_BANG_.invokeStatic(setup.clj:149)
metabase-mysql1       | 	at metabase.db.setup$run_schema_migrations_BANG_.invoke(setup.clj:144)
metabase-mysql1       | 	at metabase.db.setup$setup_db_BANG_$fn__53486$fn__53487.invoke(setup.clj:167)
metabase-mysql1       | 	at metabase.util.jvm$do_with_us_locale.invokeStatic(jvm.clj:239)
metabase-mysql1       | 	at metabase.util.jvm$do_with_us_locale.invoke(jvm.clj:225)
metabase-mysql1       | 	at metabase.db.setup$setup_db_BANG_$fn__53486.invoke(setup.clj:161)
metabase-mysql1       | 	at metabase.db.setup$setup_db_BANG_.invokeStatic(setup.clj:160)
metabase-mysql1       | 	at metabase.db.setup$setup_db_BANG_.invoke(setup.clj:153)
metabase-mysql1       | 	at metabase.db$setup_db_BANG_$fn__53511.invoke(db.clj:86)
metabase-mysql1       | 	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:81)
metabase-mysql1       | 	at metabase.db$setup_db_BANG_.doInvoke(db.clj:68)
metabase-mysql1       | 	at clojure.lang.RestFn.invoke(RestFn.java:421)
metabase-mysql1       | 	at metabase.core$init_BANG__STAR_.invokeStatic(core.clj:117)
metabase-mysql1       | 	at metabase.core$init_BANG__STAR_.invoke(core.clj:98)
metabase-mysql1       | 	at metabase.core$init_BANG_.invokeStatic(core.clj:170)
metabase-mysql1       | 	at metabase.core$init_BANG_.invoke(core.clj:165)
metabase-mysql1       | 	at metabase.core$start_normally.invokeStatic(core.clj:182)
metabase-mysql1       | 	at metabase.core$start_normally.invoke(core.clj:176)
metabase-mysql1       | 	at metabase.core$entrypoint.invokeStatic(core.clj:215)
metabase-mysql1       | 	at metabase.core$entrypoint.doInvoke(core.clj:209)
metabase-mysql1       | 	at clojure.lang.RestFn.invoke(RestFn.java:397)
metabase-mysql1       | 	at clojure.lang.AFn.applyToHelper(AFn.java:152)
metabase-mysql1       | 	at clojure.lang.RestFn.applyTo(RestFn.java:132)
metabase-mysql1       | 	at clojure.lang.Var.applyTo(Var.java:705)
metabase-mysql1       | 	at clojure.core$apply.invokeStatic(core.clj:667)
metabase-mysql1       | 	at clojure.core$apply.invoke(core.clj:662)
metabase-mysql1       | 	at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
metabase-mysql1       | 	at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
metabase-mysql1       | 	at clojure.lang.RestFn.invoke(RestFn.java:397)
metabase-mysql1       | 	at clojure.lang.AFn.applyToHelper(AFn.java:152)
metabase-mysql1       | 	at clojure.lang.RestFn.applyTo(RestFn.java:132)
metabase-mysql1       | 	at metabase.bootstrap.main(Unknown Source)
metabase-mysql1       | Caused by: java.sql.SQLSyntaxErrorException: (conn=8) Truncated incorrect INTEGER value: '0.5'
metabase-mysql1       | 	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:62)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
metabase-mysql1       | 	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
metabase-mysql1       | 	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:362)
metabase-mysql1       | 	at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:500)
metabase-mysql1       | 	at com.mchange.v2.c3p0.impl.NewProxyStatement.execute(NewProxyStatement.java:75)
metabase-mysql1       | 	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:464)
metabase-mysql1       | 	... 86 more
metabase-mysql1       | Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Truncated incorrect INTEGER value: '0.5'
metabase-mysql1       | 	at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:263)
metabase-mysql1       | 	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:356)
metabase-mysql1       | 	... 89 more
metabase-mysql1       | Caused by: java.sql.SQLException: Truncated incorrect INTEGER value: '0.5'
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:257)
metabase-mysql1       | 	... 90 more
metabase-mysql1       | 
metabase-mysql1       | UPDATE SUMMARY
metabase-mysql1       | Run:                         80
metabase-mysql1       | Previously run:             315
metabase-mysql1       | Filtered out:                 5
metabase-mysql1       | -------------------------------
metabase-mysql1       | Total change sets:          400
metabase-mysql1       | 
metabase-mysql1       | 
metabase-mysql1       | FILTERED CHANGE SETS SUMMARY
metabase-mysql1       | DBMS mismatch:                5
metabase-mysql1       | 
metabase-mysql1       | 2024-07-13 01:01:07,687 ERROR metabase.core :: Metabase Initialization FAILED
metabase-mysql1       | liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset migrations/001_update_migrations.yaml::v50.2024-06-12T12:33:09::piranha:
metabase-mysql1       |      Reason: liquibase.exception.DatabaseException: (conn=8) Truncated incorrect INTEGER value: '0.5' [Failed SQL: (1292) UPDATE cache_config
metabase-mysql1       |    SET config = json_object(
metabase-mysql1       |            'multiplier', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-ttl-ratio'), 10),
metabase-mysql1       |            'min_duration_ms', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-min-ttl'), 60000)
metabase-mysql1       |          )
metabase-mysql1       |  WHERE model = 'root' AND
metabase-mysql1       |        model_id = 0 AND
metabase-mysql1       |        strategy = 'ttl' AND
metabase-mysql1       |        (json_extract(config, '$.multiplier') = 0 OR
metabase-mysql1       |         json_extract(config, '$.min_duration_ms') = 0)]
metabase-mysql1       | 	at liquibase.command.CommandScope.execute(CommandScope.java:253)
metabase-mysql1       | 	at liquibase.Liquibase.lambda$update$0(Liquibase.java:245)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at liquibase.Liquibase.runInScope(Liquibase.java:1419)
metabase-mysql1       | 	at liquibase.Liquibase.update(Liquibase.java:234)
metabase-mysql1       | 	at liquibase.Liquibase.update(Liquibase.java:212)
metabase-mysql1       | 	at liquibase.Liquibase.update(Liquibase.java:194)
metabase-mysql1       | 	at metabase.db.liquibase$migrate_up_if_needed_BANG_$fn__44522.invoke(liquibase.clj:360)
metabase-mysql1       | 	at metabase.db.liquibase$run_in_scope_locked$reify__44518.run(liquibase.clj:325)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at metabase.db.liquibase$run_in_scope_locked.invokeStatic(liquibase.clj:318)
metabase-mysql1       | 	at metabase.db.liquibase$run_in_scope_locked.invoke(liquibase.clj:301)
metabase-mysql1       | 	at metabase.db.liquibase$migrate_up_if_needed_BANG_.invokeStatic(liquibase.clj:349)
metabase-mysql1       | 	at metabase.db.liquibase$migrate_up_if_needed_BANG_.invoke(liquibase.clj:342)
metabase-mysql1       | 	at metabase.db.setup$migrate_BANG_$fn__53458.invoke(setup.clj:84)
metabase-mysql1       | 	at metabase.db.liquibase$do_with_liquibase$f_STAR___44459.invoke(liquibase.clj:140)
metabase-mysql1       | 	at metabase.db.liquibase$do_with_liquibase.invokeStatic(liquibase.clj:143)
metabase-mysql1       | 	at metabase.db.liquibase$do_with_liquibase.invoke(liquibase.clj:131)
metabase-mysql1       | 	at metabase.db.setup$migrate_BANG_.invokeStatic(setup.clj:73)
metabase-mysql1       | 	at metabase.db.setup$migrate_BANG_.doInvoke(setup.clj:55)
metabase-mysql1       | 	at clojure.lang.RestFn.invoke(RestFn.java:425)
metabase-mysql1       | 	at metabase.db.setup$run_schema_migrations_BANG_.invokeStatic(setup.clj:149)
metabase-mysql1       | 	at metabase.db.setup$run_schema_migrations_BANG_.invoke(setup.clj:144)
metabase-mysql1       | 	at metabase.db.setup$setup_db_BANG_$fn__53486$fn__53487.invoke(setup.clj:167)
metabase-mysql1       | 	at metabase.util.jvm$do_with_us_locale.invokeStatic(jvm.clj:239)
metabase-mysql1       | 	at metabase.util.jvm$do_with_us_locale.invoke(jvm.clj:225)
metabase-mysql1       | 	at metabase.db.setup$setup_db_BANG_$fn__53486.invoke(setup.clj:161)
metabase-mysql1       | 	at metabase.db.setup$setup_db_BANG_.invokeStatic(setup.clj:160)
metabase-mysql1       | 	at metabase.db.setup$setup_db_BANG_.invoke(setup.clj:153)
metabase-mysql1       | 	at metabase.db$setup_db_BANG_$fn__53511.invoke(db.clj:86)
metabase-mysql1       | 	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:81)
metabase-mysql1       | 	at metabase.db$setup_db_BANG_.doInvoke(db.clj:68)
metabase-mysql1       | 	at clojure.lang.RestFn.invoke(RestFn.java:421)
metabase-mysql1       | 	at metabase.core$init_BANG__STAR_.invokeStatic(core.clj:117)
metabase-mysql1       | 	at metabase.core$init_BANG__STAR_.invoke(core.clj:98)
metabase-mysql1       | 	at metabase.core$init_BANG_.invokeStatic(core.clj:170)
metabase-mysql1       | 	at metabase.core$init_BANG_.invoke(core.clj:165)
metabase-mysql1       | 	at metabase.core$start_normally.invokeStatic(core.clj:182)
metabase-mysql1       | 	at metabase.core$start_normally.invoke(core.clj:176)
metabase-mysql1       | 	at metabase.core$entrypoint.invokeStatic(core.clj:215)
metabase-mysql1       | 	at metabase.core$entrypoint.doInvoke(core.clj:209)
metabase-mysql1       | 	at clojure.lang.RestFn.invoke(RestFn.java:397)
metabase-mysql1       | 	at clojure.lang.AFn.applyToHelper(AFn.java:152)
metabase-mysql1       | 	at clojure.lang.RestFn.applyTo(RestFn.java:132)
metabase-mysql1       | 	at clojure.lang.Var.applyTo(Var.java:705)
metabase-mysql1       | 	at clojure.core$apply.invokeStatic(core.clj:667)
metabase-mysql1       | 	at clojure.core$apply.invoke(core.clj:662)
metabase-mysql1       | 	at metabase.bootstrap$_main.invokeStatic(bootstrap.clj:31)
metabase-mysql1       | 	at metabase.bootstrap$_main.doInvoke(bootstrap.clj:28)
metabase-mysql1       | 	at clojure.lang.RestFn.invoke(RestFn.java:397)
metabase-mysql1       | 	at clojure.lang.AFn.applyToHelper(AFn.java:152)
metabase-mysql1       | 	at clojure.lang.RestFn.applyTo(RestFn.java:132)
metabase-mysql1       | 	at metabase.bootstrap.main(Unknown Source)
metabase-mysql1       | Caused by: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset migrations/001_update_migrations.yaml::v50.2024-06-12T12:33:09::piranha:
metabase-mysql1       |      Reason: liquibase.exception.DatabaseException: (conn=8) Truncated incorrect INTEGER value: '0.5' [Failed SQL: (1292) UPDATE cache_config
metabase-mysql1       |    SET config = json_object(
metabase-mysql1       |            'multiplier', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-ttl-ratio'), 10),
metabase-mysql1       |            'min_duration_ms', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-min-ttl'), 60000)
metabase-mysql1       |          )
metabase-mysql1       |  WHERE model = 'root' AND
metabase-mysql1       |        model_id = 0 AND
metabase-mysql1       |        strategy = 'ttl' AND
metabase-mysql1       |        (json_extract(config, '$.multiplier') = 0 OR
metabase-mysql1       |         json_extract(config, '$.min_duration_ms') = 0)]
metabase-mysql1       | 	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:151)
metabase-mysql1       | 	at liquibase.command.core.AbstractUpdateCommandStep.lambda$run$0(AbstractUpdateCommandStep.java:110)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at liquibase.command.core.AbstractUpdateCommandStep.run(AbstractUpdateCommandStep.java:108)
metabase-mysql1       | 	at liquibase.command.core.UpdateCommandStep.run(UpdateCommandStep.java:105)
metabase-mysql1       | 	at liquibase.command.CommandScope.execute(CommandScope.java:217)
metabase-mysql1       | 	... 58 more
metabase-mysql1       | Caused by: liquibase.exception.MigrationFailedException: Migration failed for changeset migrations/001_update_migrations.yaml::v50.2024-06-12T12:33:09::piranha:
metabase-mysql1       |      Reason: liquibase.exception.DatabaseException: (conn=8) Truncated incorrect INTEGER value: '0.5' [Failed SQL: (1292) UPDATE cache_config
metabase-mysql1       |    SET config = json_object(
metabase-mysql1       |            'multiplier', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-ttl-ratio'), 10),
metabase-mysql1       |            'min_duration_ms', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-min-ttl'), 60000)
metabase-mysql1       |          )
metabase-mysql1       |  WHERE model = 'root' AND
metabase-mysql1       |        model_id = 0 AND
metabase-mysql1       |        strategy = 'ttl' AND
metabase-mysql1       |        (json_extract(config, '$.multiplier') = 0 OR
metabase-mysql1       |         json_extract(config, '$.min_duration_ms') = 0)]
metabase-mysql1       | 	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:797)
metabase-mysql1       | 	at liquibase.changelog.visitor.UpdateVisitor.executeAcceptedChange(UpdateVisitor.java:119)
metabase-mysql1       | 	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:68)
metabase-mysql1       | 	at liquibase.changelog.ChangeLogIterator$2.lambda$run$0(ChangeLogIterator.java:133)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at liquibase.changelog.ChangeLogIterator$2.run(ChangeLogIterator.java:122)
metabase-mysql1       | 	at liquibase.Scope.lambda$child$0(Scope.java:186)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:195)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:185)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:164)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:252)
metabase-mysql1       | 	at liquibase.Scope.child(Scope.java:256)
metabase-mysql1       | 	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:89)
metabase-mysql1       | 	... 66 more
metabase-mysql1       | Caused by: liquibase.exception.DatabaseException: (conn=8) Truncated incorrect INTEGER value: '0.5' [Failed SQL: (1292) UPDATE cache_config
metabase-mysql1       |    SET config = json_object(
metabase-mysql1       |            'multiplier', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-ttl-ratio'), 10),
metabase-mysql1       |            'min_duration_ms', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-min-ttl'), 60000)
metabase-mysql1       |          )
metabase-mysql1       |  WHERE model = 'root' AND
metabase-mysql1       |        model_id = 0 AND
metabase-mysql1       |        strategy = 'ttl' AND
metabase-mysql1       |        (json_extract(config, '$.multiplier') = 0 OR
metabase-mysql1       |         json_extract(config, '$.min_duration_ms') = 0)]
metabase-mysql1       | 	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:470)
metabase-mysql1       | 	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:77)
metabase-mysql1       | 	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:179)
metabase-mysql1       | 	at liquibase.executor.AbstractExecutor.execute(AbstractExecutor.java:141)
metabase-mysql1       | 	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1285)
metabase-mysql1       | 	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:755)
metabase-mysql1       | 	... 81 more
metabase-mysql1       | Caused by: java.sql.SQLSyntaxErrorException: (conn=8) Truncated incorrect INTEGER value: '0.5'
metabase-mysql1       | 	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:62)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
metabase-mysql1       | 	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
metabase-mysql1       | 	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:362)
metabase-mysql1       | 	at org.mariadb.jdbc.MariaDbStatement.execute(MariaDbStatement.java:500)
metabase-mysql1       | 	at com.mchange.v2.c3p0.impl.NewProxyStatement.execute(NewProxyStatement.java:75)
metabase-mysql1       | 	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:464)
metabase-mysql1       | 	... 86 more
metabase-mysql1       | Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Truncated incorrect INTEGER value: '0.5'
metabase-mysql1       | 	at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:195)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:263)
metabase-mysql1       | 	at org.mariadb.jdbc.MariaDbStatement.executeInternal(MariaDbStatement.java:356)
metabase-mysql1       | 	... 89 more
metabase-mysql1       | Caused by: java.sql.SQLException: Truncated incorrect INTEGER value: '0.5'
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1693)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1555)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1518)
metabase-mysql1       | 	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:257)
metabase-mysql1       | 	... 90 more
metabase-mysql1       | 2024-07-13 01:01:07,692 INFO metabase.core :: Metabase Shutting Down ...
metabase-mysql1       | 2024-07-13 01:01:07,695 INFO metabase.server :: Shutting Down Embedded Jetty Webserver
metabase-mysql1       | 2024-07-13 01:01:07,708 WARN db.liquibase :: ()
metabase-mysql1       | 2024-07-13 01:01:07,713 INFO metabase.core :: Metabase Shutdown COMPLETE

```

### Information about your Metabase installation

```JSON
v50
```

### Severity

P1

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by qnkhuat**
Fixes https://github.com/metabase/metabase/issues/45520

### CODE REVIEW

**Review by crisptrutski — APPROVED** [review]

File: resources/migrations/custom_sql/fill_cache_config.my.sql [hunk 1]
```diff
@@ -4,8 +4,8 @@ WITH root AS (
          0      AS model_id,
          'ttl'  AS strategy,
          json_object(
-           'multiplier', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-ttl-ratio'), 10),
-           'min_duration_ms', coalesce((select cast(`value` as unsigned) from setting where `key` = 'query-caching-min-ttl'), 60000)
+           'multiplier', coalesce((select cast(cast(`value` as float) as unsigned) from setting where `key` = 'query-caching-ttl-ratio'), 10),
```
Comment (crisptrutski) [inline resources/migrations/custom_sql/fill_cache_config.my.sql:7]: Are you sure we want to truncate here, e.g. `2.9` becomes `2`? Can we not rather support a fractional ratio?

File: resources/migrations/custom_sql/fill_cache_config.my.sql [hunk 2]
```diff
@@ -4,8 +4,8 @@ WITH root AS (
...
+           'min_duration_ms', coalesce((select cast(cast(`value` as float) as unsigned) from setting where `key` = 'query-caching-min-ttl'), 60000)
```
Comment (crisptrutski) [inline resources/migrations/custom_sql/fill_cache_config.my.sql:8]: Minor, but I think a ceiling is safer than a truncation for this case.

Also, this expression is failing for MariaDB, but I'm not sure why.

File: resources/migrations/custom_sql/fill_cache_config.my.sql (hunk 1 above)
Comment (qnkhuat) [reply resources/migrations/custom_sql/fill_cache_config.my.sql:7]: I don't have strong opinion tbh, this is just follow the existing logic on postgres.

File: resources/migrations/custom_sql/fill_cache_config.my.sql (hunk 2 above)
Comment (qnkhuat) [reply resources/migrations/custom_sql/fill_cache_config.my.sql]: ceiled all up

**Review by crisptrutski — APPROVED** [review]
LGTM

**Comment by darksciencebase** [comment]
@qnkhuat @crisptrutski have you guys checked that it plays nice with the FE?
the PR description is missing the acceptance criteria so it's hard to tell (yes, this is a hint)

**Comment by crisptrutski** [comment]
I see the merge had a flake. I'll see about manually testing the FE issue.

**Comment by crisptrutski** [comment]
The frontend now has client side validation which rejects fractions, and if I manually hit the backend with a fractional request then I get back a somewhat-friendly Malli error which is much better than the clojure error that used to occur in that FE ticket.

**Comment by qnkhuat** [comment]
I'm not sure how this relates to FE since this is a fix for migration.
Also these settings are removed on 50.

**Pull request merged**